### ISSUE TITLE
TF2 Strange PDA and being able to use the upgrade station in MvM.

### ISSUE THREAD

**Issue by azav** [ValveSoftware/Source-1-Games#3398]
When an engineer uses the strange PDA in MvM, any time someone uses any of the engineer's equipment (teleporter, dispenser) or any time that damage to the enemy is registered, if the engineer is attempting to upgrade, he is "pushed back" from the upgrade station and is unable to upgrade.  This makes upgrading during an MvM battle very very difficult.

It appears that any property that is registered such as dispensing health, recording amount of teleports, damage caused, ammo dispenser, health awarded, all of these when registered will kick the engie back from the upgrade booth when using a Strange PDA in any MvM game.

**Comment by TrickyTrix935** [comment]
From what I've seen this appears to be linked to the code block I've placed below found in c_tf_upgrades.cpp in the games internal files:
```
		// check for other popups
		if ( bInZone && !CHudElement::ShouldDraw() ) 
		{
			CTFStatPanel *pStatPanel = GetStatPanel();
			if ( pStatPanel->IsVisible() )
			{
				pStatPanel->Hide();
				return false;
			}
		}
```
Also, this bug seems to effect not only strange PDAs but also under a few other conditions while using the upgrade station. I have listed these below:

- Clients strange weapon counts being changed
- Soul Gargoyle increases in count
- Another player leaves or joins the clients party
- "New Item" popups appearing 
- Another player leaving or joining the game

Commenting out the block I placed above should fix this issue without causing reduction of functionality to the upgrade station.

**Comment by AzureWoof** [comment]
Adding to this. There's currently an issue where bots have been created that constantly target a specific user and attempt to join off of them through the in-game party system infinitely, despite not being on their friends list. This has been purposefully done due to the fact that party updates also cause the upgrade menu in MvM to close. This stops you from being able to purchase upgrades at the upgrade station entirely and makes missions practically impossible to complete.

**Comment by jh34ghu43gu** [comment]
@AzureWoof I believe that's being tracked in #4288, although seems like (according to Trix's reply) these are both the same bug. I always explained it to people as "any game coordinator update" since the GC handles loadouts and party invites. Possibly rename this issue to "TF2 MvM Game coordinator updates prevent players from using the upgrade station (stranges and party invites)", and since it's older close the aforementioned one?

**Comment by jh34ghu43gu** [comment]
This is fixed as a side effect of them fixing #4288

**Comment by kisak-valve** [comment]
Closing per the last comment.
